packages/verified-fetch/src/plugins/plugin-handle-dag-walk.ts

```diff
@@ -0,0 +1,53 @@
+import { code as dagCborCode } from '@ipld/dag-cbor'
+import { code as dagPbCode } from '@ipld/dag-pb'
+import { handlePathWalking } from '../utils/walk-path.js'
+import { BasePlugin } from './plugin-base.js'
+import type { PluginContext } from './types.js'
+
+/**
+ * This plugin should almost always run first because it's going to handle path walking if needed, and will only say it can handle
+ * the request if path walking is possible (path is not empty, terminalCid is unknown, and the path has not been walked yet).
+ *
+ * Once this plugin has run, the PluginContext will be updated and then this plugin will return false for canHandle, so it won't run again.
+ */
+export class DagWalkPlugin extends BasePlugin {
+  /**
+   * Return false if the path has already been walked, otherwise return true if the CID is encoded with a codec that supports pathing.
+   */
+  canHandle (context: PluginContext): boolean {
+    this.log('checking if we can handle %c with accept %s', context.cid, context.accept)
+    const { pathDetails, cid } = context
+    if (pathDetails != null) {
+      // path has already been walked
+      return false
+    }
+
+    return (cid.code === dagPbCode || cid.code === dagCborCode)
```

Comment:
Do we not walk the path for dag-json?

Reply:
we didn't previously.. but we can if walking a path for dag-json makes sense and is supported.

it doesn't seem like it's currently possible when looking at https://github.com/ipfs/helia/blob/main/packages/dag-json/src/index.ts, but like it could be when looking at https://ipld.io/specs/codecs/dag-json/spec/

Reply:
fwiw there are conformance tests that aims to ensure dag-cbor an dag-json pathing behaves the same way:
- https://github.com/ipfs/gateway-conformance/blob/63f1d22c97bddf4e84b7bcc4bf1ddcea8530151a/tests/path_gateway_dag_test.go#L353